cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -620,6 +620,14 @@ else()
              "https://github.com/google/snappy/archive/${ARROW_SNAPPY_BUILD_VERSION}.tar.gz"
              "https://github.com/ursa-labs/thirdparty/releases/download/latest/snappy-${ARROW_SNAPPY_BUILD_VERSION}.tar.gz"
     )
+
+    # This can be removed when https://github.com/google/snappy/pull/148 is released
+    # Some platforms don't have patch, but this is probably ok to skip
+    find_program(patch "patch")
+    if(patch)
+      set(SNAPPY_PATCH_COMMAND "patch" "snappy.cc"
+                               "${CMAKE_SOURCE_DIR}/build-support/snappy-UBSAN.patch")
+    endif()
```

Comment:
Snappy works without the patch, though it has a UBSAN error, so this patches _if we can_ and moves on if not so we don't break.